SubZero/src/util/filter/HSVFilter.h

```diff
@@ -0,0 +1,112 @@
+/**
+  * HSVFilter.h
+  * Created on: June 19, 2015
+  * Author: James534
+  */
+
+#ifndef HSVFILTER_H
+#define HSVFILTER_H
+
+#include "Filter.h"
+#include <opencv2/imgproc.hpp>
+
+/**
+ * The HSV Filter allows filtering of Hue, Saturation, Value
+ * Uses Opencv's built-in HSV filter, along with noise filtering functions
+ */
+class HSVFilter : public Filter{
+public:
+
+    /* ==================================================
+     * Constructors
+     * ==================================================
+     */
+    /**
+     * Empty HSV filter. Set up the filter values by the set___ functions
+     */
+    HSVFilter();
+
+    /**
+     * HSVFilter with all the parameters specified
+     * @param lowH  = lower bound of Hue
+     * @param highH = upper bound of Hue
+     * @param lowS  = lower bound of Saturation
+     * @param highS = upper bound of Saturation
+     * @param lowV  = lower bound of Value
+     * @param highV = upper bound of Value
+     */
+    HSVFilter(int lowH, int highH, int lowS, int highS, int lowV, int highV);
+
+    /**
+     * HSVFilter constructor to be used when wanting to copy the filter
+     * @param obj = HSVFilter to copy
+     */
+    HSVFilter(HSVFilter* obj);
+
+    /**
+     * Operator overload
+     * @param rhs = HSVFilter to copy
+     * @return depp copy of rhs
+     */
+    HSVFilter* operator=(HSVFilter rhs);
+
+    /* =================================================
+     * Filtering
+     * =================================================
+     */
+
+    /**
+     * Filters the img in the ImgData according to the values specified
+     * Replaces the image in _data_ with the filtered image
+     * @param data = image data to be processed
+     * @return 0, successful filter
+     */
+    int filter(ImgData* data);
```

Comment:
Jame, for this filter implementation and declaration, can you refer to the TemplateFilter class? filter manager will put through a Data* as arg and we can't implicitly type cast it to ImgData*... I can't remember if it is a complier error or some other consequences during run time. In any case the declaration should have a Data* for arg and in the filter function there should be a dynamic type cast to ImgData* and a check to make sure the cast was successful. :)